### ISSUE TITLE
Schema config does not get propagated properly in some flows

### ISSUE THREAD

**Issue by bionic-ido** [ent/ent#3983]
When using a schema config to decide which tables are in which schema, in some flows of using ent, the schema config does not get propagated properly, and thus the query breaks (doesn't build parts of it with the correct schemas from the config)

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
When using a Where predicate after a QueryX, the selector it receives is without the proper schema config (when using schema configs).
E.g.:
{CODE}
query := client.Debug().TableA.Query().Where(tablea.ID(id)).
        QueryTableB().Where(
        tableb.X(x),
    ).
        QueryTableA().Where(
        service.HasTableC(),
    )
{CODE}
In this case, HasTableC receives a selector without a schema.
This happens due to line 1034 in dialect/sql/sqlgraph/graph.go from what I debugged. It seems like From is inhabited, but does not contain a context. Since it is taken as the selector without using WithContext on it, the context is empty, and thus a schema config cannot be gotten when creating the selector in HasTableA predicate.
<img width="1035" alt="image" src="https://github.com/ent/ent/assets/77567333/d344c74b-7dbc-4b19-8420-4e8f87a86518">

## Expected Behavior 🤔
The expected behavior is that it would receive and use the proper schema even in those flows, so that the generated query will be valid.
The solution seems to be either adding WithContext to the line I provided above, or otherwise making sure that the From gets generated with a proper context containing the schema config, rather then an empty context (leaning towards the first solution, as it sounds like we would want to use the current context)

## Steps to Reproduce 🕹

{CODE}
query := client.Debug().TableA.Query().Where(tablea.ID(id)).
        QueryTableB().Where(
        tableb.X(x),
    ).
        QueryTableA().Where(
        service.HasTableC(),
    )
{CODE}
causing the code generated by HasTableC to not receive its schema from the schema config.

Steps:

1.
2.
3.
4.

## Your Environment 🌎

using postgres dialect

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.21.5  |
| Ent         | v0.12.3   |
| Database    | Postgres   |
| Driver      | https://github.com/go-sql-driver/mysql        |

**Comment by bionic-ido** [comment]
Seems like there is an issue in the generated code of OrderByTableA as well, as it generates the step without regarding the schema config at all :)

**Comment by migueldv90** [comment]
I just discovered this issue too but in the Order. When I use the regular Order(classroom.ByID()) it works but when I do Order(classroom.ByID(sql.OrderDesc())) the schema fails. 

Trying to figure out a workout around it.

**Comment by bionic-ido** [comment]
yeah, I noticed that too afterwards as per my comment, though I haven't found a workaround for that one as I did with the Through issue. Seems like it is just the generated code itself that just doesn't use the schema properly there if I remember correctly, and I just left it alone since I could just avoid using it for now.
Seems like this feature needs some bugfixes, I think they are probably really small fixes in the source code of ent :)

**Comment by migueldv90** [comment]
I'm about to jump on and see if I can solve this problem, but I might have to use my backup orm (bun.uptrace) to solve this issue. There has been a few limitations that I have encountered with ENT, I had to use bun.uptrace for those case scenarios.

**Comment by bionic-ido** [comment]
Yeah, overall for places where I can't use Ent we normally just use raw sql, though most of the times it is for very specific cases where we want to optimize performance and not outright bugs like this haha, but every ORM has those I guess :)
